provider/github/watcher.go

```diff
@@ -169,6 +180,37 @@ func (w *Watcher) watchLoop(
 	}
 }
 
+// logGitHubErr logs an ErrGitHubAPI at debug or error level, depending on the
+// last time it was logged on error level
+func (w *Watcher) logGitHubErr(
+	ctx context.Context,
+	m map[*lookout.RepositoryInfo]lastErrLog,
+	repo *lookout.RepositoryInfo,
+	errMsg string,
+	resp *github.Response,
+	err error,
+) {
+	l := m[repo]
+
+	asErr := l.err != err.Error() || time.Now().After(l.last.Add(errLogInterval))
```

Comment:
I thought we want to discard every repeated error on the same repos if they happen in the same time frame, don't we?

If I'm not mistaken, with the current implementation, every new error over the same repo will clean the throttler, won't it?

example:
```go
history := make(map[*lookout.RepositoryInfo]lastErrLog)
repo1 := &lookout.RepositoryInfo{"repo-1"}
w.logGitHubErr(ctx, history, repo1, "error A", resp, "")
w.logGitHubErr(ctx, history, repo1, "error B", resp, "")
w.logGitHubErr(ctx, history, repo1, "error A", resp, "")
w.logGitHubErr(ctx, history, repo1, "error C", resp, "")
```

will log:
```
error A
error B
error A
error C
```

but it should be:
```
error A
error B
error C
```

Reply:
The current PR tries to deal with the scenario described in #385 only.

If the repo does not exist, you will get a lot of 404, and it's unlikely you will get any other error message. That could also happen if a repo you have configured suddenly becomes private and your token does not grant access to it. Or if you are offline.

For a misconfiguration (wrong repo url, wrong repo credentials, you are offline) you don't want to see the same message thousands of times.

For the scenario you describe, I can't think of a use case. If you are getting a different error each time, it's unlikely any of those errors are from a misconfiguration, and I'd say we probably want to log all of them.

Reply:
I find more usual/expected the behavior I described:
> _avoid repeated errors in a time frame_

than the implemented one:
> _avoid two errors being repeated in a row in a time frame_

On the other hand, the implementation of the first one would not be way more complicated; it would only require, for example, to store a deep `map[*lookout.RepositoryInfo]map[string]lastErrLog` instead of the current plain `map[*lookout.RepositoryInfo]lastErrLog`; or to store a plain map indexed by the mix of repo and err.